shared/src/androidMain/kotlin/com/karumi/gallery/domain/TimeProvider.kt

```diff
@@ -0,0 +1,29 @@
+package com.karumi.gallery.domain
+
+import android.content.SharedPreferences
+import java.util.Calendar
+
+actual class TimeProvider(
+  private val sharedPreferences: SharedPreferences
+) {
+  companion object {
+    private const val TIME_ARG = "time_arg"
+  }
+
+  actual fun persistTime() {
+    sharedPreferences.edit().apply {
```

Comment:
Using `apply` twice in the same method with such a different meaning hurts readability IMO. I'd use an `preferences` variable even though it's like ~9 more characters 😝

Reply:
If the problem is duplicate names and readability, I'd use `run` instead `apply` 👍